File: ViewModels/HistoryViewModel/GameDetailsHistoryView.cs
using BlackJack.ViewModels.GameViewModel.Enum;

using System.Collections.Generic;

namespace BlackJack.ViewModels.HistoryViewModel
{
    public class GameDetailsHistoryView
    {
        public List<PlayerGameDetailsHistoryViewItem> PlayersList = new List<PlayerGameDetailsHistoryViewItem>();
        public PlayerGameDetailsHistoryViewItem DealerPlayer = new PlayerGameDetailsHistoryViewItem();
        public int GameId;
    }
    public class PlayerGameDetailsHistoryViewItem
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public int CardSum { get; set; }
        public ResultEnumView Result;
        public RoleEnumView Role { get; set; }
        public List<CardGameDetailsHistoryViewItem> PlayerCards = new List<CardGameDetailsHistoryViewItem>();
    }
    public class CardGameDetailsHistoryViewItem
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public int Value { get; set; }
        public string Suit { get; set; }
    }
}

File: BlackJack.AngularMVC/models/history-model/GameDetailsHistoryView.ts
import { ResultEnumView } from "models/enum-model/ResultEnumView";
import { RoleEnumView } from "models/enum-model/RoleEnumView";
import { GameEnd } from "models/enum-model/GameEnd";

export class GameDetailsHistoryView {
    public playersList: Array<PlayerGameDetailsHistoryViewItem>;
    public dealerPlayer: PlayerGameDetailsHistoryViewItem;
    public checkEndGame: GameEnd;
    public gameId: number;
}

export class PlayerGameDetailsHistoryViewItem {
    public id: number;
    public name: string;
    public cardSum: number;
    public result: ResultEnumView;
    public role: RoleEnumView;
    public playerCards: Array<CardGameDetailsHistoryViewItem>;
}

export class CardGameDetailsHistoryViewItem {
    public id: number;
    public name: string;
    public value: number;
    public suit: string;
}